test/tests/serial/mail_hostname.bats

```diff
@@ -1,213 +1,260 @@
   assert_success
-}
-
-@test "checking configuration: non-subdomain: check hostname in postfix HELO message" {
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "nc -w 1 0.0.0.0 25 | grep domain.com"
+  # Postfix HELO message should contain FQDN (hostname)
+  _run_in_container_bash "nc -w 1 0.0.0.0 25"
+  assert_output --partial "220 ${EXPECTED_FQDN} ESMTP"
   assert_success
-}
 
-@test "checking configuration: non-subdomain: check headers of received mail" {
-  run docker exec mail_non_subdomain_hostname /bin/sh -c "ls -A /var/mail/localhost.localdomain/user1/new | wc -l | grep 1"
+  # Dovecot
+  _run_in_container_bash "doveconf hostname"
+  assert_output "hostname = ${EXPECTED_FQDN}"
   assert_success
 
-  run docker exec mail_non_subdomain_hostname /bin/sh -c "cat /var/mail/localhost.localdomain/user1/new/* | grep domain.com"
+  # OpenDMARC
+  _run_in_container_bash "grep '^AuthservID' /etc/opendmarc.conf"
+  assert_output --partial " ${EXPECTED_FQDN}"
   assert_success
-}
-
-@test "checking SRS: non-subdomain is handled correctly" {
-  docker exec mail_non_subdomain_hostname cat /etc/default/postsrsd
-  run docker exec mail_non_subdomain_hostname grep "SRS_DOMAIN=domain.com" /etc/default/postsrsd
+  _run_in_container_bash "grep '^TrustedAuthservIDs' /etc/opendmarc.conf"
+  assert_output --partial " ${EXPECTED_FQDN}"
   assert_success
-}
 
-@test "checking dovecot: non-subdomain postmaster address" {
-  run docker exec mail_non_subdomain_hostname /bin/sh -c "grep 'postmaster_address = postmaster@domain.com' /etc/dovecot/conf.d/15-lda.conf"
+  # Amavis
+  _run_in_container_bash "grep '^\$myhostname' /etc/amavis/conf.d/05-node_id"
+  assert_output "\$myhostname = \"${EXPECTED_FQDN}\";"
   assert_success
 }
 
-#
-# clean exit
-#
+function _should_have_correct_mail_headers() {
+  local EXPECTED_FQDN=${1}
+  # NOTE: The next two params should not differ for bare domains:
+  local EXPECTED_DOMAINPART=${2:-${EXPECTED_FQDN}}
+  # Required when EXPECTED_FQDN would not match the container hostname:
+  # (eg: OVERRIDE_HOSTNAME or `--hostname mail --domainname example.test`)
+  local EXPECTED_HOSTNAME=${3:-${EXPECTED_FQDN}}
 
-@test "checking that the container stops cleanly: mail_override_hostname" {
-  run docker stop -t 60 mail_override_hostname
+  _run_in_container_bash "nc 0.0.0.0 25 < /tmp/docker-mailserver-test/email-templates/existing-user1.txt"
   assert_success
-}
 
-@test "checking that the container stops cleanly: mail_non_subdomain_hostname" {
-  run docker stop -t 60 mail_non_subdomain_hostname
-  assert_success
-}
+  _wait_for_empty_mail_queue_in_container
+  _count_files_in_directory_in_container '/var/mail/localhost.localdomain/user1/new/' '1'
 
-@test "checking that the container stops cleanly: mail_srs_domainname" {
-  run docker stop -t 60 mail_srs_domainname
+  # MTA hostname (sender?) is used in filename of stored mail:
+  local MAIL_FILEPATH="$(_exec_in_container_bash "ls -A /var/mail/localhost.localdomain/user1/new")"
```

Comment:
This really seems awkward to me. Rember with `ls`: it is designed to serve output _humans_. ShellCheck warns: "parsing output from `ls` may produce unstable results". Maybe we can first do this:

```suggestion
    local MAIL_FILEPATH="$(_exec_in_container ls -A /var/mail/localhost.localdomain/user1/new)
```

and then something like

```suggestion
    local MAIL_FILEPATH="$(_exec_in_container find /var/mail/localhost.localdomain/user1/new -type f -maxdepth 1)
```

Not sure if the second command really is equivalent; please double-check.

Reply:
I'll look into it, this is just what the original test had and seemed to be used in other tests too.

Reply:
Your last suggestion works well, thanks :)